src/memo/memo.ml

```diff
@@ -145,9 +145,15 @@ module Id = Id.Make ()
 module Caches = struct
   let cleaners = ref []
 
-  let register ~clear = cleaners := clear :: !cleaners
+  let invalidators = ref []
+
+  let register ~clear ~invalidate =
+    cleaners := clear :: !cleaners;
+    invalidators := invalidate :: !invalidators
 
   let clear () = List.iter !cleaners ~f:(fun f -> f ())
+
+  let invalidate () = List.iter !cleaners ~f:(fun f -> f ())
```

Comment:
Bug: iterating over `cleaners` here instead of over `invalidators`.

Reply:
Oops, well spotted :)

Reply:
Done.